### ISSUE TITLE
Improve the code style by introducing macros

### ISSUE THREAD

**Issue by git-hulk** [apache/kvrocks#2521]
Followup code improvements mentioned in PR #2508. Currently, we check and return error if the status is not ok in many many places, which can be improved by introducing macros.

The naming candidates can be:

- RETURN_NOT_OK
- RETURN_IF_ERROR

and for converting the rocksdb status if there is an error:

- RETURN_NOT_OK_FROM_ROCKSDB
- RETURN_IF_ERR_FROM_ROCKSDB

_Originally posted by @mapleFU in https://github.com/apache/kvrocks/pull/2508#pullrequestreview-2276461263_

**Comment by git-hulk** [comment]
@AntiTopQuark Would you like to take this issue?

**Comment by AntiTopQuark** [comment]
> @AntiTopQuark Would you like to take this issue?

I’d be happy to take on this issue.  ：）

**Comment by PragmaTwice** [comment]
The first macro is duplicated with the current GET_OR_RET macro.

**Comment by PragmaTwice** [comment]
For the second I think it depends on the context, e.g. rocksdb::Status -> Status? rocksdb::Status -> rocksdb::Status?

We can extend GET_OR_RET for the latter, and the former doesn't require new macro.

**Comment by git-hulk** [comment]
> For the second I think it depends on the context, e.g. rocksdb::Status -> Status? rocksdb::Status -> rocksdb::Status?
> 
> We can extend GET_OR_RET for the latter, and the former doesn't require new macro.

Yes, my bad. GET_OR_RET allows ignoring the return value, so the former one is unnecessary.
